tareas/clase-6/tarea-clase-6.js

```diff
@@ -5,12 +5,164 @@ Al hacer click en "calcular", mostrar en un elemento pre-existente la mayor edad
 
 Punto bonus: Crear un botón para "empezar de nuevo" que empiece el proceso nuevamente, borrando los inputs ya creados (investigar cómo en MDN).
 */
+document.querySelector("[id=mostrar_formularios]").onclick = function() {
+ 
+    let cantidadIntegrantes = document.querySelector("#cantidad_integrantes").value;
+    //validar cantidad integrantes
+    if (cantidadIntegrantes <= 0){
+      alert("La cantidad de integrantes no puede ser menor a 0");
+      return false;
+    }
+    if (cantidadIntegrantes > 10){
+      alert("Lo siento, la cantidad de integrantes no puede ser mayor a 10");
+      return false;
+    }
+
+    //disablo el button
+    document.getElementById("mostrar_formularios").disabled = true;
+    desplegarFormularioEdades(cantidadIntegrantes);    
+
+}
+
+function desplegarFormularioEdades(cantidadIntegrantes){
+    let nuevoInput = "";
+    let nuevoParrafo = "";
+    let textoParrafo = "";
+   
+    let divisor = ""; 
+    const nodoPagina = document.querySelector("[id=formulario_ingreso_edades]");
+  
+    for (let i = 1; i <= cantidadIntegrantes; i++) {
+      
+        divisor = document.createElement("div");
+        divisor.setAttribute("class", "integrante");
+
+        nuevoParrafo = document.createElement("label");
+        textoParrafo = document.createTextNode("Edad del integrante: ");
+        nuevoParrafo.appendChild(textoParrafo);
+
+        nuevoInput = document.createElement("input");
+        nuevoInput.setAttribute("type", "number");
+        nuevoInput.setAttribute("id", "edadIntegrante" + i);
+        
+        divisor.appendChild(nuevoParrafo);
+        divisor.appendChild(nuevoInput);
+        nodoPagina.appendChild(divisor);
+    }
+}
+ 
+//ACLARACION MIA: ESTO FUNCIONA PERO VOY A HACERLO COMO EN EL CURSO
+/*
+let i=1;
+ while (i<= cantidadIntegrantes) {
+var y = document.createElement("INPUT");
+y.setAttribute("type", "text");
+y.setAttribute("Placeholder", "Name_" + i);
+y.setAttribute("Name", "Name_" + i);
+document.getElementById("formulario_ingreso_edades").appendChild(y);
+i++;
+}
+*/
+
+
+
+document.querySelector("[id=calculo_edades]").onclick = function() {
+
+//validar edades
+let cajas = document.querySelector("[id=formulario_ingreso_edades]").querySelectorAll("input");
+let senial = 0;
+
+for (i=0; i<cajas.length;i++){
+  
+  if (cajas[i].value <1 || cajas[i].value > 120){
+    alert("Debe ingresar una edad entre 1 y 120");
+    senial = 1;
+		return false;
```

Comment:
Otra cosa que te puede ayudar a subir el nivel de tu código es recurrir a los métodos de un array. En este caso `Array.every()` ejecutaría el mismo procedimiento que estás ejecutando con este for y te ahorra varias líneas de código [Aquí](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Array/every) puedes estudiarlos